src/utils.ts

```diff
@@ -0,0 +1,71 @@
+import fs from "fs";
+import path from "path";
+import sharp from "sharp";
+
+const __dirname = path.dirname(new URL(import.meta.url).pathname);
+const cacheFolderPath = path.join(__dirname, "..", "cache");
+const cacheImagesFolderPath = path.join(cacheFolderPath, "images");
+const cacheThumbnailsFolderPath = path.join(cacheFolderPath, "thumbnails");
+
+export function createCacheFolders() {
+  if (!fs.existsSync(cacheFolderPath)) {
+    fs.mkdirSync(cacheFolderPath);
+  }
+
+  if (!fs.existsSync(cacheImagesFolderPath)) {
+    fs.mkdirSync(cacheImagesFolderPath);
+  }
+
+  if (!fs.existsSync(cacheThumbnailsFolderPath)) {
+    fs.mkdirSync(cacheThumbnailsFolderPath);
+  }
+}
+
+export async function saveProfilePicture(
+  userId: string,
+  pictureBuffer: Buffer
+) {
+  const originalFilePath = path.join(cacheImagesFolderPath, `${userId}.jpeg`);
+  const thumbnailFilePath = path.join(
+    cacheThumbnailsFolderPath,
+    `${userId}_thumbnail.jpeg`
+  );
+  try {
+    await sharp(pictureBuffer).resize(100, 100).toFile(thumbnailFilePath);
+
+    console.log("Successfully saved profile picture and thumbnail.");
+
+    fs.writeFileSync(originalFilePath, pictureBuffer);
+  } catch (error) {
+    console.error("Error saving profile picture:", error);
+    fs.writeFileSync(originalFilePath, pictureBuffer);
+  }
+}
+
+export function serveCachedImage(imageType: string, userId: string, res: any) {
+  const cacheFolderPath =
+    imageType === "images" ? cacheImagesFolderPath : cacheThumbnailsFolderPath;
+
+  const fileName =
+    imageType === "images"
+      ? `${userId}.jpeg`
+      : imageType === "thumbnails"
+      ? `${userId}.jpeg`
+      : null;
+
+  if (!fileName) {
```

Comment:
I think if the fileName is null, in line 63, no file will be found?

Reply:
Yes?

Reply:
Then you maybe don't need the null check before? ;-)